ide/services/ide-editor/src/main/java/org/eclipse/dirigible/runtime/ide/editor/services/EditorRestService.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Objects;
+
+/**
+ * Front facing REST service serving the Editor content.
+ */
+@Path("/ide/editor")
+@Api(value = "IDE - Editor", authorizations = {@Authorization(value = "basicAuth", scopes = {})})
+@ApiResponses({@ApiResponse(code = 401, message = "Unauthorized"), @ApiResponse(code = 403, message = "Forbidden")})
+public class EditorRestService extends AbstractRestService implements IRestService {
+
+    private static final Logger logger = LoggerFactory.getLogger(EditorRestService.class);
+
+    private final IRepository repository = (IRepository) StaticObjects.get(StaticObjects.REPOSITORY);
+
+    @Context
+    private HttpServletResponse response;
+
+    /**
+     * Finds all prettier config file paths inside a project
+     *
+     * @param workspaceName the workspace
+     * @param projectName   the project to search the config file in
+     * @param filePath      the config file name with extension
+     * @return the found prettier config paths
+     */
+    @GET
+    @Path("/prettier/config/{workspaceName}/{projectName}/{path:.*}")
+    public Response findPrettierConfig(
+            @PathParam("workspaceName") String workspaceName,
+            @PathParam("projectName") String projectName,
+            @PathParam("path") String filePath,
+            @Context HttpServletRequest request) {
+        String user = UserFacade.getName();
+        if (user == null) {
+            return createErrorResponseForbidden(NO_LOGGED_IN_USER);
+        }
+
+        String configFileName = ".prettierrc.json";
+        String configContentType = "application/json";
+        String separator = IRepository.SEPARATOR;
+
+        String projectPath = joinPathParts(separator, "users", user, workspaceName, projectName);
+        String queryPath = joinPathParts(separator, "prettier", "config", workspaceName, projectName, filePath);
+
+        RepositoryPath sourceFilePath = new RepositoryPath(projectPath + IRepository.SEPARATOR + filePath);
+        RepositoryPath sourceFolder = sourceFilePath.getParentPath();
+        String sourceFileName = sourceFilePath.getLastSegment();
+        ICollection lookupDirectory = repository.getCollection(sourceFolder.getPath());
+
+        if (!isPathNormalized(queryPath) || !isResourceInDirectory(lookupDirectory, sourceFileName)) {
+            return createErrorResponseNotFound(
+                    logMessage("There is resource at the specified path", queryPath)
+            );
+        }
+
+        while (isSubPath(projectPath, lookupDirectory.getPath())) {
+            String content = getConfigFileContent(lookupDirectory, configFileName);
+            if(content != null && !content.isEmpty()) {
+                return Response.ok(content).type(configContentType).build();
+            }
+
+            lookupDirectory = lookupDirectory.getParent();
+        }
+
+        return createErrorResponseNotFound(
+                logMessage("There is no config file for resource", queryPath)
+        );
```

Comment:
Also, do we actually need the query path being logged into the error message?

Reply:
I agree that the `logMessage` method seems redundant. @Fluctuationqt maybe you could still reuse code if you create a method `createPrettierConfigNotFoundErrorResponse` and make it log before returning an HTTP response? But if you think the `queryPath` parameter could be skipped, this would be redundant as well.